pkg/rke2/config.go

```diff
@@ -356,6 +364,96 @@ func newRKE2ServerConfig(opts ServerConfigOpts) (*rke2ServerConfig, []bootstrapv
 		rke2ServerConfig.CloudControllerManagerExtraEnv = opts.ServerConfig.CloudControllerManager.ExtraEnv
 	}
 
+	// POC: Add system-node agent install script, config and connection info as files
+	serverUrlSetting := &unstructured.Unstructured{}
+	serverUrlSetting.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "management.cattle.io",
+		Kind:    "Setting",
+		Version: "v3",
+	})
+	if err := opts.Client.Get(context.Background(), client.ObjectKey{
+		Name: "server-url",
+	}, serverUrlSetting); err != nil {
+		return nil, nil, fmt.Errorf("failed to get server url setting: %w", err)
+	}
+	serverUrl, ok := serverUrlSetting.Object["value"].(string)
+	if !ok {
+		return nil, nil, fmt.Errorf("server url setting is missing value")
+	}
+
+	caSetting := &unstructured.Unstructured{}
+	caSetting.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "management.cattle.io",
+		Kind:    "Setting",
+		Version: "v3",
+	})
+	if err := opts.Client.Get(context.Background(), client.ObjectKey{
+		Name: "cacerts",
+	}, caSetting); err != nil {
+		return nil, nil, fmt.Errorf("failed to get ca setting: %w", err)
+	}
+
+	pem, ok := caSetting.Object["value"].(string)
+	if !ok {
+		return nil, nil, fmt.Errorf("ca setting is missing value")
+	}
+
+	kubeConfig, err := clientcmd.Write(clientcmdapi.Config{
+		Clusters: map[string]*clientcmdapi.Cluster{
+			"agent": {
+				Server:                   serverUrl,
+				CertificateAuthorityData: []byte(pem),
+			},
+		},
+		AuthInfos: map[string]*clientcmdapi.AuthInfo{
+			"agent": {
+				Token: string(opts.ServiceAccountToken),
+			},
+		},
+		Contexts: map[string]*clientcmdapi.Context{
+			"agent": {
+				Cluster:  "agent",
+				AuthInfo: "agent",
+			},
+		},
+		CurrentContext: "agent",
+	})
+
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate kubeconfig: %w", err)
+	}
+
+	connectInfoConfig := struct {
+		Namespace  string `json:"namespace"`
+		SecretName string `json:"secretName"`
+		KubeConfig string `json:"kubeConfig"`
+	}{
+		Namespace:  opts.Machine.Namespace,
+		SecretName: opts.MachinePlanSecretName,
+		KubeConfig: string(kubeConfig),
+	}
+
+	connectInfoConfigJson, err := json.MarshalIndent(connectInfoConfig, "", " ")
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to marshal connect info config: %w", err)
+	}
+
+	files = append(files, bootstrapv1.File{
+		Path:        "/etc/rancher/agent/connect-info-config.json",
```

Comment:
does it have to match with https://github.com/rancher/system-agent/tree/main/examples/configuration file name: `conninfo.yaml`

Reply:
not necessarily, in CAPR they have a similar JSON